packages/core/src/components/tabs2/tabs2.tsx

```diff
@@ -150,9 +150,15 @@ export class Tabs2 extends AbstractComponent<ITabs2Props, ITabs2State> {
         }
     }
 
-    public componentDidUpdate(_prevProps: ITabs2Props, prevState: ITabs2State) {
+    public componentDidUpdate(prevProps: ITabs2Props, prevState: ITabs2State) {
         if (this.state.selectedTabId !== prevState.selectedTabId) {
             this.moveSelectionIndicator();
+        } else if (prevState.selectedTabId != null) {
+            const selectedTabChildren = this.getTabChildren();
+            const prevSelectedTabChildren = this.getTabChildren(prevProps.children);
+            if (selectedTabChildren !== prevSelectedTabChildren) {
```

Comment:
This is not an easy issue to solve, but I came with a different way that might work. I compare the title, id, length and position of the tabs in the current and previous state, if they are not identical then move the indicator. Let me know what do you think about that fix.